src/Compat.jl

```diff
@@ -51,6 +51,15 @@ end
     end
 end
 
+# https://github.com/JuliaLang/julia/pull/43852
+if VERSION < v"1.8.0-DEV.1484"
```

Comment:
```suggestion
@static if VERSION < v"1.8.0-DEV.1484"
```
should make this work even if `@assume_effects` is undefined.

Reply:
No, scratch that, it's just that the two branches of the `if` need to be swapped, right? Before `v"1.8.0-DEV.1484"`, we want to define our own macro, after we use the one from `Base`. Not the other way round.